doc/api/packages.md

```diff
@@ -689,6 +689,17 @@ exports, while resolving the existing `"node"`, `"node-addons"`, `"default"`,
 
 Any number of custom conditions can be set with repeat flags.
 
+Conditions can be almost any string, including multi-byte characters and
+whitespace. There's only a few restrictions:
```

Comment:
I would prefer that we do not say this, and instead say something less specific, eg that special characters are permitted in condition names. We should probably ban spaces IMO.

Reply:
We could start this with a stronger recommendation (e.g. keep it to `/^[a-z0-9:=-]+$/`) and only afterwards go into the technical side of what's enforced?

I'm +1 on banning whitespace and also banning quotes and non-ASCII characters. I don't see a clear use case for those but could see many confusing bugs. Conditions usually fail silently, so debugging a "looked like character A but was actually character B" problem seems super annoying.

Reply:
Added a more opinionated opening paragraph before delving into the technical restrictions.